lib/apiservers/service/restapi/handlers/vch_log_get.go

```diff
@@ -0,0 +1,200 @@
+	}
+
+	output, err := getContentFromLogFile(op.Context, helper, logFilePaths)
+	if err != nil {
+		return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogDefault(util.StatusCode(err)).WithPayload(&models.Error{Message: err.Error()})
+	}
+
+	return operations.NewGetTargetTargetDatacenterDatacenterVchVchIDLogOK().WithPayload(output)
+}
+
+// getDatastoreHelper validates the VCH and returns the datastore helper for the VCH. It errors when validation fails or when datastore is not ready
+func getDatastoreHelper(ctx context.Context, d *data.Data) (*datastore.Helper, error) {
+	// TODO (angiew): abstract some of the boilerplate into helpers in common.go
+	validator, err := validateTarget(ctx, d)
+	if err != nil {
+		return nil, util.WrapError(400, err)
+	}
+
+	executor := management.NewDispatcher(validator.Context, validator.Session, nil, false)
+	vch, err := executor.NewVCHFromID(d.ID)
+	if err != nil {
+		return nil, util.NewError(500, fmt.Sprintf("failed to create VCH %s: %s", d.ID, err))
+	}
+
+	err = validate.SetDataFromVM(validator.Context, validator.Session.Finder, vch, d)
+	if err != nil {
+		return nil, util.NewError(500, fmt.Sprintf("Failed to load VCH data: %s", err))
+	}
+
+	// Get VCH configuration
+	vchConfig, err := executor.GetNoSecretVCHConfig(vch)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve VCH information: %s", err)
+	}
+
+	// Relative path of datastore folder
+	vmPath := vchConfig.ImageStores[0].Path
+
+	// Get VCH datastore object
+	ds, err := validator.Session.Finder.Datastore(validator.Context, vchConfig.ImageStores[0].Host)
+	if err != nil {
+		return nil, util.NewError(404, fmt.Sprintf("Datastore folder not found for VCH %s: %s", d.ID, err))
+	}
+
+	// Create a new datastore helper for file finding
+	helper, err := datastore.NewHelper(ctx, validator.Session, ds, vmPath)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to get datastore helper: %s", err)
+	}
+
+	return helper, nil
+}
+
+// getAllLogFilePaths returns a list of all log file paths under datastore folder, errors out when no log file found
+func getAllLogFilePaths(ctx context.Context, helper *datastore.Helper) ([]string, error) {
+	res, err := helper.Ls(ctx, "")
+	if err != nil {
+		return nil, fmt.Errorf("Unable to list all files under datastore: %s", err)
+	}
+
+	var paths []string
+	for _, f := range res.File {
+		path := f.GetFileInfo().Path
+		if strings.HasPrefix(path, LogFilePrefix) {
+			paths = append(paths, path)
+		}
+	}
+
+	if len(paths) == 0 {
+		return nil, util.NewError(404, "No log file available in datastore folder")
+	}
+
+	return paths, nil
+}
+
+// getContentFromLogFile downloads all log files in the list, concatenates the content of each log file and outputs a string of contents
+func getContentFromLogFile(ctx context.Context, helper *datastore.Helper, paths []string) (string, error) {
+	var buffer bytes.Buffer
+
+	for _, p := range paths {
```

Comment:
Can we sort these paths before iterating through them so that we always get the operations in the order that they happened?

Reply:
The log files are automatically sorted by timestamp in datastore, and ls() preserves the order. 
But I should probably double check anyways... I'll add a sort function